### ISSUE TITLE
Failing brian2 test: `test_timestep_function_during_run`

### ISSUE THREAD

**Issue by denisalevi** [brian-team/brian2cuda#126]
`brian2.tests.test_functions:test_timestep_function_during_run` is failing with
```
AssertionError: 
Arrays are not equal

(mismatch 80.0%)
 x: array([5, 5, 5, 5, 5], dtype=int32)
 y: array([5, 6, 7, 8, 9])
```

Couldn't spot the error, moving on for now. Could be a bug in the `StateMonitor` which is recording those arrays (maybe related to #50) or a bug in the `timestep` function (see #125).

**Comment by denisalevi** [comment]
The `Network` and all its `Clock`s are running on the host and we only copy the time variables (`t`, `timestep`, `dt`) once in the beginning and once in the end of a run, not at every timestep. Therefore the device variable for `t` is zero, which results in the result above.

Easy fix: Pass `t` as kernel parameter when `timestep` is used.

**Comment by denisalevi** [comment]
I removed clock synchronization (host to device) at every clock cycle in 658925b9 to reduce overhead due to `Memcpy` for small networks. I tried to pass time variables by value whenever they are needed and changed the template code in (at least the rate-) monitor to tread the variables as values not as pointers. But this is only happening for `ArrayVariables`, not for `Function` type variables such as `timestep` or `TimedArray`.
